components/automate-ui/src/app/modules/user/user-details/user-details.component.ts

```diff
@@ -55,7 +56,12 @@ export class UserDetailsComponent implements OnInit, OnDestroy {
       filter(identity),
       takeUntil(this.isDestroyed))
       .subscribe((id: string) => {
-        this.store.dispatch(new GetUser({ id }));
+        if (this.preUrlUserId === '') {
```

Comment:
I'd prefer a fix that prevents the extra call to the user details API when we go back to the team-details page instead of letting it go through with an old ID. 

It appears that when we click the back button, the routeParams subscription on the user-details components grabs the team ID and passes it to the GetUser action.
I'm wondering if we can add another step of filtering to this block that checks the contents of the ActivatedRoute ([more info on that here](https://angular.io/guide/router#activated-route)). Then we could ensure that GetUser is dispatched only if the route includes `users`.

Let me know if you have questions about that idea!

Reply:
I concur with @bcmdarroch and her suggested approach is a good one and should work. However, I think we can simplify a bit further by considering the context of the situation.
What Brenna correctly describes is that the route change is firing an Observable on the page we are in the process of leaving; we have no need to do that. Thinking further, we really only need to fire this subscription exactly once--when we arrive on the page.
So adding the `first()` operator should address the scenario:
```
    this.store.pipe(
      select(routeParams),
      pluck('id'),
      filter(identity),
      first())
      .subscribe((id: string) => {
          this.store.dispatch(new GetUser({ id }));
      });
```